### DISCUSSION TITLE
Timer during handshake

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#195] [Project Q&A 2. Transport Layer (TCP)]
Hello.

When implementing handshake, I confirmed the server receives SYN packet and send SYN/ACK.
I thought that after the server sends SYN, packetArrived() should be called in the client side as the client will receive SYN/ACK.
However, the program execution do not enter packetArrived() function again, resulting in a unknown file Failure error.

I set the timer when sending SYN/ACK in the server side. 
Are there any more operations should I implement when entering packetArrived() for SYN/ACK in client side?

Thanks in advance.

**Comment by Pusnow** [comment]
> I thought that after the ~server~ client sends SYN, packetArrived() should be called in the ~client~ server side as the ~client~ server will receive SYN/ACK.
> However, the program execution do not enter packetArrived() function again, resulting in a unknown file Failure error.

This is because we test your TCP implementation with TA's implementation. If the server uses your implementation, the client uses ours, and vice versa.

Or there is a possibility for header error. If you do not fill headers correctly (especially destination IP address and checksum), packets can be dropped.

> Are there any more operations should I implement when entering packetArrived() for SYN/ACK in client side?

Sending SYN/ACK is enough. However, depending on implementations, you may need to create a TCB (TCP control block) to maintain initiating connection.
